conjure-java-core/src/integrationInput/java/com/palantir/product/AliasAsMapKeyExample.java

```diff
@@ -95,6 +95,11 @@ public boolean equals(Object other) {
     }
 
     private boolean equalTo(AliasAsMapKeyExample other) {
+        if (this.memoizedHashCode != 0
+                && other.memoizedHashCode != 0
+                && this.memoizedHashCode != other.memoizedHashCode) {
+            return false;
+        }
```

Comment:
this is a decent representative example of the value of avoiding recomputing `equals` for each of the `Map` fields